src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const expensesService = require('../services/expenses.services');
+const userService = require('../services/users.services');
+const { expensesFilter } = require('../utils/expensesFilter');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+    categories,
+  } = req.query;
+
+  const expenses = expensesFilter(userId, from, to, categories);
+
+  res.send(expenses);
+};
+
+const getById = (req, res) => {
+  const { id } = req.params;
+  const searcedExpense = expensesService.getById(+id);
+
+  if (!id) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  if (!searcedExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.status(200).send(searcedExpense);
+};
+
+const post = (req, res) => {
+  const { userId, ...rest } = req.body;
+
+  if (!userService.getById(userId)) {
+    return res.sendStatus(400);
+  }
+
+  const newExpense = {
+    id: Date.now(),
```

Comment:
I am not sure whether it is a good idea to add date as id, there are some libs that generate id. Besides you may just write the simple id generator
```
const idGenerator = (arr) => {
  if (!arr.length) {
    return 1;
  }

  const maxId = Math.max(...arr.map(({ id }) => id));

  return maxId + 1;
};
```